src/utils/formatters.ts

```diff
@@ -0,0 +1,199 @@
+/*
+  This module contains utilites for formatting values in visualizations.
+  
+  A `Formatter` takes a value in a table and formats it as a user-facing
+  string. A formatter factory creates a `Formatter`. Here we define several
+  formatter factories for common use cases such as formatting time values, or
+  formatting quantities of bytes.
+
+      // Create a formatter by calling the factory
+      const myTimeFormatter = timeFormatter({timezone: 'America/New_York'})
+
+      // Use the formatter to get a user-friendly string representation of a value
+      myTimeFormatter(1562018849599)
+
+      // => returns "1 July 2019"
+
+*/
+
+import {format as d3Format} from 'd3-format'
+
+import {Formatter} from '../types'
+
+export const TIME_FORMATTER_TYPE: 'TIME' = 'TIME'
+
+interface TimeFormatter extends Formatter {
+  (x: number, options?: {domainWidth?: number}): string
+
+  _GIRAFFE_FORMATTER_TYPE: typeof TIME_FORMATTER_TYPE
+}
+
+interface TimeFormatterFactoryOptions {
```

Comment:
We are using the browser [`Intl`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DateTimeFormat) API for zero-dependency time zone and locale support.